pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -541,6 +541,41 @@ public SuccessResponse resetSegment(
     }
   }
 
+  /**
+   * Resets segments with Error EV of the given table
+   * This API will take segments to OFFLINE state, wait for External View to stabilize, and then back to
+   * ONLINE/CONSUMING state,
+   * thus effective in resetting segments or consumers in error states.
+   */
+  @POST
+  @Path("segments/{tableNameWithType}/resetErrorEvSegments")
```

Comment:
Suggest modifying `segments/{tableNameWithType}/reset` and include a query parameter `errorSegmentsOnly` instead of introducing a new API since their purpose is very similar

Reply:
make sense, updated